doc/api/packages.md

```diff
@@ -607,22 +608,28 @@ Runtimes or tools other than Node.js can use them at their discretion.
 
 These user conditions can be enabled in Node.js via the [`--conditions` flag][].
 
-The following condition definitions are currently endorsed by Node.js:
+The following condition definitions are currently endorsed by Node.js, and
+are _listed in order from most to least specific, as conditions should be
+used._
 
-* `"browser"` - any environment which implements a standard subset of global
-  browser APIs available from JavaScript in web browsers, including the DOM
-  APIs.
+* `"types"` - can be used by typing systems to resolve the typing file for
+  the given export, possible since the interface should be the same for all
+  variations. _This condition should always be included first._
+* `"deno"` - indicates a variation for the Deno platform.
+* `"browser"` - any web browser environment.
 * `"development"` - can be used to define a development-only environment
-  entry point. _Must always be mutually exclusive with `"production"`._
+  entry point, for example to provide additional debugging context such as
+  better error messages when running in a development mode. _Must always be
+  mutually exclusive with `"production"`._
 * `"production"` - can be used to define a production environment entry
   point. _Must always be mutually exclusive with `"development"`._
 
 The above user conditions can be enabled in Node.js via the
 [`--conditions` flag][].
 
-Platform specific conditions such as `"deno"`, `"electron"`, or `"react-native"`
-may be used, but while there remain no implementation or integration intent
-from these platforms, the above are not explicitly endorsed by Node.js.
+Platform-specific conditions such as `"electron"`, or `"react-native"` may
+be used, but while there remain no implementation or integration intent
+from these platforms, they are not currently explicitly recommended.
```

Comment:
Optional nit. I think this might be more clear:

```suggestion
from these platforms, we do not recommended them.
```

Reply:
In professional documentation, I expect the passive voice. It’s not clear who the “we” would be here.

Reply:
The use of "we recommend" is from the Microsoft Style Guide, which we've adopted so that we can keep out own local style guide nice and short. (It only needs a few parochial things like making sure we use _Node.js_ and not _NodeJS_.)

While they don't sound enthusiastic about it, they explicitly say "we recommend" is OK, and explicitly say to avoid passive voice "it is recommended". Ref: https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/r/recommend

The reasoning is probably the reasoning in their [Three voice principles](https://docs.microsoft.com/en-us/style-guide/brand-voice-above-all-simple-human#three-voice-principles):

> Our voice is:
> 
> * Warm and relaxed—We’re natural. Less formal, more grounded in real, everyday conversations. Occasionally, we’re fun. (We know when to celebrate.)

Reply:
To me, it's reasonably clear that "we" here refers to the Node.js project. However, if you want to be a bit more formal, you can remove the "recommend"/"endorse" stuff entirely and be imperative:

```suggestion
from these platforms, avoid them when possible.
```

...or something along those lines.

Reply:
All of that said, I'm OK if my nit here is ignored. While I'd _prefer_ one of my suggestions be adopted so we have consistent style in accordance with our chosen style guide etc., I'm definitely more interested in not burning out contributors on purely stylistic doc nits!

Reply:
The current version was passive, which is why it stuck out to me that this was proposing changing it. Perhaps the greater issue here is that this is very clearly an opinion; why should the user “avoid them when possible,” for example? Maybe we should take an extra sentence or two and explain why the user should do the thing a certain way, and then we don’t need make a recommendation or express an opinion. Something like:

```suggestion
Platform-specific conditions such as `"electron"`, or `"react-native"` may
be used when the target is unavoidably a single runtime; but maximum
compatibility with all potential consumers can be achieved by using broader
targets such as `"import"`.
```

Reply:
Taking a step back, this entire section is actually redundant. We already note the application-specific non-listing in the requirements for custom definition listings. This section was originally written before we had other custom definitions in place, so was initiall more about giving examples of what conditions might be used for beyond the Node.js ones. Since this is no longer necessary and is adding more confusion I've gone ahead and removed it.